### ISSUE TITLE
New users should be directed to join or create a team before going to the messages screen

### ISSUE THREAD

**Issue by johnneed** [johnneed/GreenUpVermont#71]
Our nav should default to the teams page if the user has no teams, otherwise it should default to the messages screen (current behavior)

**Comment by johnneed** [comment]
FYI the TabNavigator takes an initialRouteName config setting.  You could count the user's teams and if it's zero do something like this

initialRouteName :  usersTeams.length > 0 ? 'Messages' : 'Teams';

**Comment by johnneed** [comment]
Here's the docs for the TabNavigator 
https://reactnavigation.org/docs/tab-navigator.htm

**Comment by smaraf** [comment]
@johnneed  I thought that was going to work and I did try it but for some reason, it's not as straightforward as it looks. I did get it to work by passing in the initialRouteName but then ran into another 2 issues:

- Tab Navigator did not re-render properly once teams loaded so it would always default to the Teams screen because no teams were loaded when the navigator was first rendered - I could have probably get around this by messing with the state but then there was the second issue that was even more puzzling
- The Stack Navigation stopped working altogether when I tried passing in a dynamic initial route name to the TabNavigator - this must be due to the way the navigation props are passed in but I could not figure it out. 

I ended up going with a less elegant but working solution where the Messages screen is the initial route and when it loads it will navigate to the Teams screen if the logged in user doesn't have any teams.
